web/src/components/metrics/histogram.tsx

```diff
@@ -113,17 +130,74 @@ export const Histogram: React.FC<{
     [moveHistogramRange, moveRange, zoomRange]
   );
 
-  const zoomButtonTips = () => {
+  const zoomButtonTips = React.useCallback(() => {
     return t('Zoom in / out histogram. You can also use plus or minus buttons while histogram is focused.');
-  };
+  }, [t]);
 
-  const pageButtonTips = () => {
+  const pageButtonTips = React.useCallback(() => {
     return t('Move selected range. You can also use page up or down buttons while histogram is focused.');
-  };
+  }, [t]);
 
-  const arrowButtonTips = () => {
+  const arrowButtonTips = React.useCallback(() => {
     return t('Move displayed range. You can also use arrow left or right buttons while histogram is focused.');
-  };
+  }, [t]);
+
+  const [guidedTourDone, setGuidedTourDone] = useLocalStorage<boolean>(LOCAL_STORAGE_HISTOGRAM_GUIDED_TOUR_DONE_KEY);
+  React.useEffect(() => {
+    if (!guidedTourHandle) {
+      return;
+    }
+
+    guidedTourHandle.updateTourItems([
+      {
+        title: t('Histogram'),
+        description: t(
+          // eslint-disable-next-line max-len
+          'The following bar chart represent the number of logs across time. You can select a portion of it to drill down into the selected time range and filter the content below accordingly.'
+        ),
+        assetName: 'histogram.gif',
+        minWidth: '500px',
+        ref: containerRef
+      },
+      {
+        title: t('Zoom buttons'),
+        description: zoomButtonTips(),
+        assetName: 'histogram-zoom.gif',
+        position: PopoverPosition.bottom,
+        ref: zoomRef
+      },
+      {
+        title: t('Arrow buttons'),
+        description: arrowButtonTips(),
+        ref: arrowRef
+      },
+      {
+        title: t('Page buttons'),
+        description: pageButtonTips(),
+        assetName: 'histogram-pages.gif',
+        minWidth: '500px',
+        position: PopoverPosition.bottom,
+        ref: pageRef
+      }
+    ]);
+
+    if (!guidedTourDone) {
+      setGuidedTourDone(true);
+      guidedTourHandle.startTour();
+    }
```

Comment:
Every component using the guided tour can call `updateTourItems` to define a tour (and keep refs updated even if the view is removed from the dom) and `startTour` to open it from the begining using the `guidedTourHandle` provided as prop.

Reply:
nice 👍